templates/admin/user/edit.tmpl

```diff
@@ -116,7 +116,7 @@
 						<input name="restricted" type="checkbox" {{if .User.IsRestricted}}checked{{end}}>
 					</div>
 				</div>
-				<div class="inline field">
+				<div class="inline field"{{if DisableGitHooks}} hidden{{end}}>
```

Comment:
```suggestion
				<div class="inline field{{if DisableGitHooks}} hidden{{end}}">
```

Reply:
Adding `hidden` class just grays out this option and we want to hide it. Using hidden div attribute

https://stackoverflow.com/questions/1992114/how-do-you-create-a-hidden-div-that-doesnt-create-a-line-break-or-horizontal-sp/27548863#27548863

works ok and hides this option which makes UI cleaner.

Reply:
Didn't know about this boolean attribute, but it looks like it has a unwanted interaction with CSS's `display` property:

> Changing the value of the CSS display property on an element with the hidden attribute overrides the behavior. For instance, elements styled display: flex will be displayed despite the hidden attribute's presence.

So elements can still show with the `hidden` attribute if there happens to be a `display` value set in CSS and for that reason, I'd prefer a class-based approach. I suggest adding a `.hidden { display: none !important }` helper in `web_src/less/helpers.less`.

Reply:
It seems that `<div hidden>` is better.

1. `.hidden` may conflict with Fomantic UI's `.hidden` CSS class.

2. Fomantic UI already has the `[hidden]` style:
<img width="684" alt="image" src="https://user-images.githubusercontent.com/2114189/151903494-45755d2a-db5a-405a-a38e-84f1f3a6ac2d.png">

Reply:
We can use `hidden` attribute, but I think that style needs to be added as an `!important` variant in our CSS, otherwise fomantic stuff will conflict, (they often set `display: block` on high-specificity selectors).

Reply:
Since it works now, we do not need to add it now. `!important` is evil, it usually means bad design or bad usage.

Reply:
`!important` is an absolute necessity for helper classes that are potentially overriden by more specific selectors, some in our control, some not.

Reply:
@silverwind 

None of these `!important` is necessary, in most cases, the `!important` usage in Gitea code is wrong.

* https://github.com/go-gitea/gitea/pull/18319#issuecomment-1015239886
* https://github.com/go-gitea/gitea/pull/17866#issuecomment-983404274
* https://github.com/go-gitea/gitea/pull/17562/commits/ab6db78982636ff9467bfd40af5f7d7c778daf3b
* https://github.com/go-gitea/gitea/pull/17898#issuecomment-997203009

Reply:
We're getting off-topic again, but I'm still of the opinion that helper-style single classes absolutely need `important` to work. There is no other way.